src/Util/Lock.php

```diff
@@ -21,191 +19,79 @@
  */
 class Lock
 {
-	private static $semaphore = [];
-
 	/**
-	 * @brief Check for memcache and open a connection if configured
-	 *
-	 * @return object|boolean The memcache object - or "false" if not successful
+	 * @var Lock\ILockDriver;
 	 */
-	private static function connectMemcache()
-	{
-		if (!function_exists('memcache_connect')) {
-			return false;
-		}
-
-		if (!Config::get('system', 'memcache')) {
-			return false;
-		}
+	static $driver = null;
 
-		$memcache_host = Config::get('system', 'memcache_host', '127.0.0.1');
-		$memcache_port = Config::get('system', 'memcache_port', 11211);
-
-		$memcache = new Memcache;
-
-		if (!$memcache->connect($memcache_host, $memcache_port)) {
-			return false;
+	public static function init()
+	{
+		switch(Config::get('system', 'lock_driver', 'default')) {
```

Comment:
I don't like this `default` default value. The current behavior resembles more to `auto` with lock driver auto-detection.